client/lib/api/user_preferences.dart

```diff
@@ -92,8 +95,26 @@ class UserPreferences {
   }
 
   Future<bool> setNotificationsEnabled(bool value) async {
-    return (await SharedPreferences.getInstance())
+    var result = (await SharedPreferences.getInstance())
         .setBool(UserPreferenceKey.NotificationsEnabled.toString(), value);
+    await _updateFirebaseAutoInit();
+    return result;
+  }
+
+  Future<void> _updateFirebaseAutoInit() async {
+    // Updates the firebase auto init option. This method is called
+    // after a user modifies either analytics or notifications.
+    var analyticsEnabled = await getAnalyticsEnabled();
+    var notificationsEnabled = await getNotificationsEnabled();
```

Comment:
@epicfaace - this compresses 6 lines of logic down to two. I think it's also much easier to read and debug:

```suggestion
    var firebase_auto_init = (await getNotificationsEnabled()) || (await getAnalyticsEnabled());
    // Enable Firebase Auto Init iff either notifications or analytics is enabled
    await firebaseMessaging.setAutoInitEnabled(firebase_auto_init);
```